src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1163,36 +1170,17 @@ mobile_apis::Result::eType RegisterAppInterfaceRequest::CheckWithPolicyData() {
     }
   }
 
-  mobile_apis::Result::eType result = mobile_apis::Result::SUCCESS;
+  auto result = mobile_apis::Result::SUCCESS;
 
   // If AppHMIType is not included in policy - allow any type
-  if (!app_hmi_types.empty()) {
-    if (message[strings::msg_params].keyExists(strings::app_hmi_type)) {
-      // If AppHmiTypes are partially same, the system should allow those listed
-      // in the policy table and send warning info on missed values
-      smart_objects::SmartArray app_types =
-          *(message[strings::msg_params][strings::app_hmi_type].asArray());
-
-      std::string log;
-      CheckMissedTypes checker(app_hmi_types, log);
-      std::for_each(app_types.begin(), app_types.end(), checker);
-      if (!log.empty()) {
-        response_info_ =
-            "Following AppHmiTypes are not present in policy "
-            "table:" +
-            log;
-        result_code_ = mobile_apis::Result::WARNINGS;
-      }
-    }
-    // Replace AppHmiTypes in request with values allowed by policy table
-    message[strings::msg_params][strings::app_hmi_type] =
-        smart_objects::SmartObject(smart_objects::SmartType_Array);
-
-    smart_objects::SmartObject& app_hmi_type =
-        message[strings::msg_params][strings::app_hmi_type];
-
-    AppHMITypeInserter inserter(app_hmi_type);
-    std::for_each(app_hmi_types.begin(), app_hmi_types.end(), inserter);
+  if (!app_hmi_types_in_policy.empty()) {
+    result = ProcessingAppHMITypesPolicies(message, app_hmi_types_in_policy);
+  } else {
+    result = ProcessingAppHMITypesInMessage(
+        message,
+        std::string("WEB_VIEW AppHmiType is absent in application policies, "
```

Comment:
@SNiukalov @IGapchuk I thought we do this check for all apps, not only for WEB_VIEW. Not clear why do we need this info string here

Reply:
@LitvinenkoIra Because WEB VIEW absense is critical-> app won't be registered.  WebView type has a logic which differs from the other App HMI types.

Reply:
@AByzhynar I understand that we need this info in the message, but I do not think that we need to pass it as a parameter here

Reply:
@AByzhynar @LitvinenkoIra 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/49/commits/594f1c62e074e20ac698d1cf7cc7b35b200d9e0d